src/NuGet.Core/NuGet.Configuration/Settings/Settings.cs

```diff
@@ -548,19 +548,14 @@ private static ISettings LoadSettingsGivenSettingsFiles(List<SettingsFile> setti
                 // If the default user config NuGet.Config does not exist, we need to create it.
                 var defaultSettingsFilePath = Path.Combine(userSettingsDir, DefaultSettingsFileName);
 
+                var defaultSettingsFilePathExistsPreviously = File.Exists(defaultSettingsFilePath);
+
                 SettingsFile userSpecificSettings = ReadSettings(rootDirectory, defaultSettingsFilePath, settingsLoadingContext: settingsLoadingContext);
-                if (File.Exists(defaultSettingsFilePath) && userSpecificSettings.IsEmpty())
+                if (!defaultSettingsFilePathExistsPreviously && File.Exists(defaultSettingsFilePath) && userSpecificSettings.IsEmpty())
                 {
```

Comment:
Can you check this line 554 again? It looks `!defaultSettingsFilePathExistsPreviously` and `File.Exists(defaultSettingsFilePath)` cancel each other so it'll always result false. Not sure if it's intended logic.

Reply:
Thanks @erdembayar for reviewing!
Line 553 will create a default config file if it doesn't exist.

Reply:
Probably worth adding a comment, as it's not obvious.

Reply:
Thanks! Added a comment.

Reply:
Just another thought if `!defaultSettingsFilePathExistsPreviously` is true then `userSpecificSettings.IsEmpty()` would be always true. Do we need to check `userSpecificSettings.IsEmpty()`?

Reply:
Thanks @erdembayar !
I just found the default config file is created if it doesn't exist in `ReadSettings` already at [here ](https://github.com/NuGet/NuGet.Client/blob/dev/src/NuGet.Core/NuGet.Configuration/Settings/SettingsFile.cs#L114)
So this whole paragraph is dead code so it could be removed.
Sorry for my misunderstanding before.